src/lib/storage/table.cpp

```diff
@@ -94,19 +94,39 @@ Table::Table(const TableColumnDefinitions& column_definitions, const TableType t
   if constexpr (HYRISE_DEBUG) {
     const auto chunk_count = _chunks.size();
     const auto num_columns = column_count();
+
+    // Used to check for referenced tables of segments of each column.
+    auto common_referenced_tables = std::vector<std::shared_ptr<const Table>>(num_columns);
+
     for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
       const auto chunk = get_chunk(chunk_id);
       if (!chunk) {
         continue;
       }
 
       Assert(chunk->size() > 0 || (type == TableType::Data && chunk_id == chunk_count - 1 && chunk->is_mutable()),
-             "Empty chunk other than mutable chunk at the end was found");
-      Assert(chunk->has_mvcc_data() == (_use_mvcc == UseMvcc::Yes), "Supply MvccData for Chunks iff Table uses MVCC");
-      Assert(chunk->column_count() == num_columns, "Invalid Chunk column count");
+             "Empty chunk other than mutable chunk at the end was found.");
+      Assert(chunk->has_mvcc_data() == (_use_mvcc == UseMvcc::Yes), "Supply MvccData for Chunks iff Table uses MVCC.");
+      Assert(chunk->column_count() == num_columns, "Invalid Chunk column count.");
 
       for (auto column_id = ColumnID{0}; column_id < num_columns; ++column_id) {
-        Assert(chunk->get_segment(column_id)->data_type() == column_data_type(column_id), "Invalid Segment DataType");
+        const auto& segment = chunk->get_segment(column_id);
```

Comment:
@dey4ss : I haven't found any checks in this direction anywhere. Even though we quietly assume it in several places.